public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx

```diff
@@ -50,6 +69,7 @@ export const TimeSeriesPanel: React.FC<TimeSeriesPanelProps> = ({
       width={width}
       height={height}
       legend={options.legend}
+      onBeforeRerender={onBeforeRerender}
```

Comment:
:thinking: it would be nice to have this measure be easy to apply to any panel.  Perhaps we can add a callback in panelContext?

Reply:
as I understand the current structure, we would need to define the callback in https://github.com/grafana/grafana/blob/54af57b8e66fcb398a4c3f7b7b1036de7292c4dd/public/app/features/dashboard/dashgrid/PanelChrome.tsx#L66 but the actual implementation of `onBeforeRerender` depends on the type of the panel. Not sure if that's the right approach unless we can maybe modify the context somewhere in the component tree?